core/app/models/spree/order.rb

```diff
@@ -364,6 +366,18 @@ def associate_user!(user, override_email = true)
       end
     end
 
+    def disassociate_user!
+      nullified_attributes = ASSOCIATED_USER_ATTRIBUTES.index_with(nil)
+
+      self.assign_attributes(nullified_attributes)
+
+      changes = slice(*ASSOCIATED_USER_ATTRIBUTES)
+
+      ActiveRecord::Base.connected_to(role: :writing) do
+        self.class.unscoped.where(id: self).update_all(changes)
```

Comment:
@caiodsc Is there a particular reason here for skipping the order callbacks/validations?